index-handlers/src/test/java/no/sikt/nva/nvi/index/apigateway/SearchNviCandidatesHandlerIntegrationTest.java

```diff
@@ -412,87 +431,155 @@ void shouldFilterByRejectedByOthers() {
           .containsExactly(tuple(GlobalApprovalStatus.DISPUTE, REJECTED));
     }
 
-    private static List<Approval> getAllPossibleApprovals(URI topLevelOrganization) {
-      var baseApproval = randomApprovalBuilder(topLevelOrganization);
-      return Stream.of(ApprovalStatus.values())
-          .map(status -> baseApproval.withApprovalStatus(status).build())
-          .toList();
+    @ParameterizedTest
+    @MethodSource("defaultAggregationCountProvider")
+    void shouldAggregateBasedOnGlobalStatusByDefault(String aggregationField, int expectedCount) {
+      var response = handleRequest(emptyMap());
+
+      var actualCount = getAggregationCount(response, aggregationField);
+      assertEquals(expectedCount, actualCount);
     }
 
-    private static List<NviCandidateIndexDocument> createDocsForAllApprovalStatusCombinations() {
-      var ourApprovals = getAllPossibleApprovals(OUR_ORGANIZATION);
-      var theirApprovals = getAllPossibleApprovals(THEIR_ORGANIZATION);
+    @ParameterizedTest
+    @MethodSource("organizationApprovalStatusAggregationProvider")
+    void shouldAggregateByOrganizationStatusOnlyWhenSet(
+        String aggregationField, int expectedCount) {
+      var response = handleRequest(Map.of("aggregationType", "organizationApprovalStatuses"));
+
+      var ourApprovals = getOrganizationSummary(response);
+      var actualCount = ourApprovals.at(aggregationField).asInt();
+      assertEquals(expectedCount, actualCount);
+    }
+
+    private static int getAggregationCount(
+        PaginatedSearchResult<NviCandidateIndexDocument> response, String field) {
+      return response.getAggregations().get(field).get("docCount").asInt();
+    }
+
+    private static JsonNode getOrganizationSummary(
+        PaginatedSearchResult<NviCandidateIndexDocument> response) {
+      return response
+          .getAggregations()
+          .get("organizationApprovalStatuses")
+          .get(OUR_ORGANIZATION.toString())
+          .get("organizations")
+          .get(OUR_ORGANIZATION.toString());
+    }
```

Comment:
The output JSON when using this query parameter is very weird and I suspect the aggregation is needlessly complex. Leaving it as-is for now, but we should revisit this.